From 91c9ae44752dd93f71e7230f2c9b1387119416a9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:20 -0400
Subject: [x86] perf: Fix Userspace RDPMC switch

Message-id: <1408953210-26343-13-git-send-email-jolsa@redhat.com>
Patchwork-id: 88732
O-Subject: [PATCH RHEL7.1 BZ1133083 012/322] perf/x86: Fix Userspace RDPMC switch
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 0e9f2204cfa6d79abe3e525ddf7c4ab5792cc751
Author: Peter Zijlstra <peterz@infradead.org>
Date: Wed Feb 5 11:19:56 2014 +0100

description
===========
The current code forgets to change the CR4 state on the current CPU.
Use on_each_cpu() instead of smp_call_function().
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/cpu/perf_event.c b/arch/x86/kernel/cpu/perf_event.c
index 5bfc466..d33b1d4 100644
--- a/arch/x86/kernel/cpu/perf_event.c
+++ b/arch/x86/kernel/cpu/perf_event.c
@@ -1824,7 +1824,7 @@ static ssize_t set_attr_rdpmc(struct device *cdev,
 
  if (!!val != !!x86_pmu.attr_rdpmc) {
   x86_pmu.attr_rdpmc = !!val;
-  smp_call_function(change_rdpmc, (void *)val, 1);
+  on_each_cpu(change_rdpmc, (void *)val, 1);
  }
 
  return count;
-- 
1.7.1